File: C++/Program/Sivitas.cpp
/* 
    Saya Hestina Dwi Hartiwi NIM 2108077 mengerjakan soal Latihan 2 dalam Praktikum
    mata kuliah Desain dan Pemrograman Berorientasi Objek untuk keberkahan-Nya maka 
    saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamin.
*/

/* import library */
#include <string>
#include <iostream>
#include "Human.cpp"

using namespace std;

/*--- Deklarasi Class ---*/
class Sivitas : public Human {
    /* Private Atributes */
    private:
        string asal_univ, email_edu;
    public:
        /* Konstruktor */
        Sivitas ()
        {
            this->asal_univ = "";
            this->email_edu = "";
        }

        /* Getter and Setter */

        // Get asal_univ
        string get_asal_univ()
        {
            return this->asal_univ;
        }

        // Get email_edu
        string get_email_edu()
        {
            return this->email_edu;
        }

        // Set asal_univ
        void set_asal_univ(string asal_univ)
        {
            this->asal_univ = asal_univ;
        }

        // Set email_edu
        void set_email_edu(string email_edu)
        {
            this->email_edu = email_edu;
        }

        void printSivitas()
        {
            cout << "Home University    : " << this->asal_univ << '\n';
            cout << "Email Edu          : " << this->email_edu << '\n';
        }

        /* Destructor */
        ~Sivitas()
        {

        }
};

File: Python/Program/Sivitas.py
# /* 
#     Saya Hestina Dwi Hartiwi NIM 2108077 mengerjakan soal Latihan 2 dalam Praktikum
#     mata kuliah Desain dan Pemrograman Berorientasi Objek untuk keberkahan-Nya maka 
#     saya tidak melakukan kecurangan seperti yang telah dispesifikasikan. Aamin.
# */

from Human import Human

class Sivitas(Human):
    __asal_univ = ""
    __email_edu = ""

    # Konstruktor
    def __init__(self):
        self.__asal_univ = ""
        self.__email_edu = ""
    
    # Getter and Setter #
    
    # Get asal_univ
    def get_asal_univ (self):
        return self.__asal_univ
    
    # Get email_edu
    def get_email_edu (self):
        return self.__email_edu

    # Set asal_univ
    def set_asal_univ (self, asal_univ):
        self.__asal_univ = asal_univ
    
    # set email_edu
    def set_email_edu (self, email_edu):
        self.__email_edu = email_edu
    
    # Public Method #
    def printSivitas(self):
        print("Home University  : ", self.get_asal_univ())
        print("Email Edu        : ", self.get_email_edu())